### PR TITLE
OCPBUGS-26940: Include OperatorHubSpec sync with HC when Config is empty

### PR DESCRIPTION
**Pull request by jparrill**
This PR Contains:
- [x] Fixed bug
- [x] Added ReconcileOLM unitTests

Signed-off-by: Juan Manuel Parrilla Madrid <jparrill@redhat.com>

**Which issue(s) this PR fixes** :
Fixes #[OCPBUGS-26940](https://issues.redhat.com/browse/OCPBUGS-26940)

### CODE REVIEW

File: control-plane-operator/hostedclusterconfigoperator/controllers/resources/resources_test.go [hunk 1]
```diff
@@ -161,6 +161,111 @@ func TestReconcileErrorHandling(t *testing.T) {
...
+			want:                &configv1.OperatorHubSpec{},
+		},
+		{
+			name: "PlacementStrategy is guest and allDefaultSources disabled",
+			hcpClusterConfig: &hyperv1.ClusterConfiguration{
+				OperatorHub: &configv1.OperatorHubSpec{
+					DisableAllDefaultSources: true,
+				},
+			},
+			olmCatalogPlacement: hyperv1.ManagementOLMCatalogPlacement,
```
Comment (bryan-cox) [inline control-plane-operator/hostedclusterconfigoperator/controllers/resources/resources_test.go:222]: Shouldn't this be guest placement?

File: control-plane-operator/hostedclusterconfigoperator/controllers/resources/resources_test.go [hunk 2]
```diff
@@ -161,6 +161,111 @@ func TestReconcileErrorHandling(t *testing.T) {
...
+			},
+		},
+		{
+			name: "PlacementStrategy is guest and allDefaultSources enabled",
+			hcpClusterConfig: &hyperv1.ClusterConfiguration{
+				OperatorHub: &configv1.OperatorHubSpec{
+					DisableAllDefaultSources: false,
+				},
+			},
+			olmCatalogPlacement: hyperv1.ManagementOLMCatalogPlacement,
```
Comment (bryan-cox) [inline control-plane-operator/hostedclusterconfigoperator/controllers/resources/resources_test.go:234]: Shouldn't this be guest placement?

File: control-plane-operator/hostedclusterconfigoperator/controllers/resources/resources.go [hunk 3]
```diff
@@ -1414,7 +1414,12 @@ func (r *reconciler) reconcileOLM(ctx context.Context, hcp *hyperv1.HostedContro
 		// Management OLM Placement
 		if _, err := r.CreateOrUpdate(ctx, r.client, operatorHub, func() error {
 			if hcp.Spec.Configuration != nil && hcp.Spec.Configuration.OperatorHub != nil {
+				// if the spec.Configuration contains something related with OperatorHub configuration
+				// then we need to sync the OperatorHub.Spec to the spec.Configuration.OperatorHub
 				operatorHub.Spec.DisableAllDefaultSources = hcp.Spec.Configuration.OperatorHub.DisableAllDefaultSources
+			} else if hcp.Spec.Configuration == nil || hcp.Spec.Configuration.OperatorHub == nil {
```
Comment (muraee) [inline control-plane-operator/hostedclusterconfigoperator/controllers/resources/resources.go:1420]: I don't think you need `else if` with a condition, `else` would be enough.

File: control-plane-operator/hostedclusterconfigoperator/controllers/resources/resources.go [hunk 4]
```diff
@@ -1414,7 +1414,12 @@ func (r *reconciler) reconcileOLM(ctx context.Context, hcp *hyperv1.HostedContro
 		// Management OLM Placement
 		if _, err := r.CreateOrUpdate(ctx, r.client, operatorHub, func() error {
 			if hcp.Spec.Configuration != nil && hcp.Spec.Configuration.OperatorHub != nil {
+				// if the spec.Configuration contains something related with OperatorHub configuration
+				// then we need to sync the OperatorHub.Spec to the spec.Configuration.OperatorHub
```
Comment (muraee) [inline control-plane-operator/hostedclusterconfigoperator/controllers/resources/resources.go:1418]: nit: could be better rephrased, something like: `if spec.Configuration.OperatorHub is set, we need to sync it to the guest cluster`. I find this comment tricky to follow at first glance.

File: control-plane-operator/hostedclusterconfigoperator/controllers/resources/resources_test.go (hunk 1 above)
Comment (jparrill) [reply control-plane-operator/hostedclusterconfigoperator/controllers/resources/resources_test.go:222]: Oh you're right, fixing it

File: control-plane-operator/hostedclusterconfigoperator/controllers/resources/resources_test.go (hunk 2 above)
Comment (jparrill) [reply control-plane-operator/hostedclusterconfigoperator/controllers/resources/resources_test.go:234]: Oh you're right, fixing it

**Pull request merged**